internal/provider/resource_api_key_test.go

```diff
@@ -795,6 +982,24 @@ func testAccCheckCloudApiKeyConfig(mockServerUrl, cloudApiKeyResourceLabel, clou
 	`, mockServerUrl, cloudApiKeyResourceLabel, cloudApiKeyDisplayName, cloudApiKeyDescription, ownerId, ownerApiVersion, ownerKind)
 }
 
+func testAccCheckTableflowApiKeyConfig(mockServerUrl, tableflowApiKeyResourceLabel, tableflowApiKeyDisplayName, tableflowApiKeyDescription, ownerId, ownerApiVersion, ownerKind, apiKeyTypeNonResourceSpecific string) string {
+	return fmt.Sprintf(`
+	provider "confluent" {
+		endpoint = "%s"
+	}
+	resource "confluent_api_key" "%s" {
+		display_name = "%s"
+		description = "%s"
+		owner {
+			id = "%s"
+			api_version = "%s"
+			kind = "%s"
+		}
+		api_key_type_non_resource_specific = "%s"
```

Comment:
I'm not sure we want to add a name with such an attribute here (i.e., api_key_type_non_resource_specific). Could you share some of the other options we considered?

Note: I understand it's a tricky situation since we sort of hardcoded an empty resource block for cloud API Keys and non-empty API Keys for SR/Kafka API Keys, etc., but I'd like to see the full thought process here.

Reply:
I have prepared a Slack message sent to our group with better context:
https://confluent.slack.com/archives/C075H73ATEX/p1735001060176429

Reply:
An alternative approach is to treat the `tableflow` api-key as if the resource specific api-key, in that case we may have to modify many existing logic inside:
https://github.com/confluentinc/terraform-provider-confluent/blob/master/internal/provider/resource_api_key.go#L387-L429
AND
https://github.com/confluentinc/terraform-provider-confluent/blob/master/internal/provider/resource_api_key.go#L275-L303

I feel this approach will be more error-prone compared to the proposed one described in the Slack message.